week04/API/src/pages/login.tsx

```diff
@@ -0,0 +1,76 @@
+import { Link, useNavigate } from 'react-router-dom';
+import { useState } from 'react';
+import { signIn } from '../api/authApi';
+import { PostLoginRequest } from '../types/Auth';
+
+const login = () => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  const navigate = useNavigate();
+  const handleLogin = async (e: React.FormEvent) => {
+    e.preventDefault();
+    try {
+      const loginData: PostLoginRequest = {
+        username,
+        password,
+      };
+      const response = await signIn(loginData);
+
+      if ('result' in response) {
+        localStorage.setItem('accessToken', response.result.token);
+        console.log('로그인 요청결과:', response); // 요청 결과 콘솔로 출력
+        alert('로그인이 완료되었습니다');
+        navigate('/hobby');
+      } else if (response.code === '01') {
+        alert('비밀번호가 틀렸습니다.');
+      }
+    } catch (error) {
+      alert('로그인 요청 중 오류가 발생했습니다');
+    }
+  };
+
+  return (
+    <div className='flex flex-col justify-center items-center min-h-screen'>
+      <main className='w-[30rem] bg-white flex flex-col justify-center items-center p-8 rounded-lg gap-4 shadow-md'>
+        <header>
+          <h1 className='w-full text-3xl font-bold mb-4'>로그인</h1>
+        </header>
+        <form className='flex flex-col gap-3 w-full'>
+          <input
+            type='text'
+            id='username'
+            name='username'
+            placeholder='아이디'
+            value={username}
+            onChange={(e) => setUsername(e.target.value)}
+            className='w-full p-5 rounded-lg border border-textSecondary placeholder:text-base text-base'
+          />
+          <input
+            type='password'
+            id='password'
+            name='password'
+            placeholder='비밀번호'
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
+            className='w-full p-5 rounded-lg border border-textSecondary placeholder:text-base text-base'
+          />
+
+          <button
+            type='submit'
+            onClick={handleLogin}
+            className='w-full p-5 rounded-lg border border-textSecondary text-lg bg-secondary text-white font-bold transition-colors  hover:bg-point'
+          >
+            로그인
+          </button>
+
```

Comment:
```suggestion
        <form className='flex flex-col gap-3 w-full' onSubmit={handleLogin}>
          <input
            type='text'
            id='username'
            name='username'
            placeholder='아이디'
            value={username}
            onChange={(e) => setUsername(e.target.value)}
            className='w-full p-5 rounded-lg border border-textSecondary placeholder:text-base text-base'
          />
          <input
            type='password'
            id='password'
            name='password'
            placeholder='비밀번호'
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            className='w-full p-5 rounded-lg border border-textSecondary placeholder:text-base text-base'
          />

          <button
            className='w-full p-5 rounded-lg border border-textSecondary text-lg bg-secondary text-white font-bold transition-colors  hover:bg-point'
          >
            로그인
          </button>

```

일단 button의 기본 type이 submit이라서 `type='submit'`이라서, 명시적으로 다시 설정해줄 필요가 없을 것 같구요!
form안에 있는 button의 submit 동작에 대한 이벤트 핸들링은 button의 onClick 보다는 form의 onSubmit으로 처리되는것이 더 좋은 방식일 것 같습니다!